crates/re_viewer/src/ui/welcome_screen/welcome_page.rs

```diff
@@ -60,114 +42,183 @@ pub(super) fn welcome_page_ui(
+            image: &re_ui::icons::WELCOME_SCREEN_EXAMPLES,
+            add_buttons: Box::new(|ui: &mut egui::Ui| {
+                large_text_button(ui, "View Examples").clicked()
+            }),
+        },
+    ];
+
+    const MAX_COLUMN_WIDTH: f32 = 255.0;
+    const MIN_COLUMN_WIDTH: f32 = 164.0;
+
+    let grid_spacing = egui::vec2(12.0, 16.0);
+
+    let mut column_count = (((ui.available_width() + grid_spacing.x)
+        / (MIN_COLUMN_WIDTH + grid_spacing.x))
+        .floor() as usize)
+        .clamp(1, panels.len());
+
+    // disallow 3 columns
+    if column_count == 3 {
+        column_count = 2;
+    }
 
-        ui.end_row();
+    let column_width = ((ui.available_width() + grid_spacing.x) / column_count as f32
+        - grid_spacing.x)
+        .floor()
+        .at_most(MAX_COLUMN_WIDTH);
 
-        ui.horizontal(|ui| {
-            button_centered_label(ui, "Quick start…");
-            // TODO(ab): activate when C++ is ready!
-            // url_large_text_button(re_ui, ui, "C++", CPP_QUICKSTART);
-            url_large_text_button(re_ui, ui, "Python", PYTHON_QUICKSTART);
-            url_large_text_button(re_ui, ui, "Rust", RUST_QUICKSTART);
-        });
+    // this space is added on the left so that the grid is centered
+    let centering_space = (ui.available_width()
+        - column_count as f32 * column_width
+        - (column_count - 1) as f32 * grid_spacing.x)
+        .max(0.0)
+        / 2.0;
 
-        ui.horizontal(|ui| {
-            if large_text_button(ui, "Open file…").clicked() {
-                command_sender.send_ui(re_ui::UICommand::Open);
-            }
-            button_centered_label(ui, "Or drop a file anywhere!");
-        });
+    ui.horizontal(|ui| {
+        ui.add_space(centering_space);
 
-        ui.horizontal(|ui| {
-            url_large_text_button(re_ui, ui, "Learn about Views", SPACE_VIEWS_HELP);
-        });
+        ui.vertical(|ui| {
+            ui.horizontal_wrapped(|ui| {
+                ui.add(egui::Label::new(
+                    egui::RichText::new("Welcome.")
+                        .strong()
+                        .line_height(Some(32.0))
+                        .text_style(re_ui::ReUi::welcome_screen_h1()),
+                ));
 
-        ui.end_row();
-    });
+                ui.add(egui::Label::new(
+                    egui::RichText::new("Visualize multimodal data.")
+                        .line_height(Some(32.0))
+                        .text_style(re_ui::ReUi::welcome_screen_h1()),
+                ));
+            });
+
+            ui.add_space(32.0);
+
+            let grid = egui::Grid::new("welcome_screen_grid")
+                .spacing(grid_spacing)
+                .min_col_width(column_width)
+                .max_col_width(column_width);
+
+            grid.show(ui, |ui| {
+                let mut show_example = false;
+
+                for panels in panels.chunks(column_count) {
+                    if column_count == 4 {
```

Comment:
magic constant

Reply:
did you mean to use `panels.len()` instead?

Reply:
This whole "try to do web-looking stuff with egui" generally feels like "magic code", but yeah here `len()` is a tad clearer 👍🏻